jumanji/environments/routing/sokoban/env.py

```diff
@@ -0,0 +1,587 @@
+                action,
+            ),
+        )
+
+        return updated_action
+
+    def update_box_push_action(
+        self,
+        fixed_grid: chex.Array,
+        variable_grid: chex.Array,
+        new_location: chex.Array,
+        action: chex.Array,
+    ) -> chex.Array:
+        """
+        Masks actions to -1 if pushing the box is not a valid move. If it
+        would be pushed out of the grid or the resulting square
+        is either a wall or another box.
+
+        Args:
+            fixed_grid: Array (uint8) shape (num_rows, num_cols) The fixed grid.
+            variable_grid: Array (uint8) shape (num_rows, num_cols) The
+            variable grid.
+            new_location: Array (int32) shape (2,) The new location of the agent.
+            action: Array (int32) shape () The action to be executed.
+
+        Returns:
+            updated_action: Array (int32) shape () The updated action after
+            checking if pushing the box is a valid move.
+        """
+
+        return jax.lax.select(
+            self.check_space(
+                variable_grid,
+                new_location + MOVES[action].squeeze(),
+                BOX,
+            )
+            | ~self.in_grid(new_location + MOVES[action].squeeze()),
+            -jnp.ones(shape=(), dtype=jnp.int32),
+            jax.lax.select(
+                self.check_space(
+                    fixed_grid,
+                    new_location + MOVES[action].squeeze(),
+                    WALL,
+                ),
+                -jnp.ones(shape=(), dtype=jnp.int32),
+                action,
+            ),
+        )
+
+    def move_agent(
+        self,
+        variable_grid: chex.Array,
+        action: chex.Array,
+        current_location: chex.Array,
+    ) -> Tuple[chex.Array, chex.Array]:
+        """
+        Executes the movement of the agent specified by the action and
+        executes the movement of a box if present at the destination.
+
+        Args:
+            variable_grid: Array (uint8) shape (num_rows, num_cols)
+            action: Array (int32) shape () The action to take.
+            current_location: Array (int32) shape (2,)
+
+        Returns:
+            next_variable_grid: Array (uint8) shape (num_rows, num_cols)
+            next_location: Array (int32) shape (2,)
+        """
+
+        next_location = current_location + MOVES[action]
+        box_location = next_location + MOVES[action]
+
+        # remove agent from current location
+        next_variable_grid = variable_grid.at[tuple(current_location)].set(EMPTY)
+
+        # either move agent or move agent and box
+        next_variable_grid = jax.lax.cond(
+            self.check_space(variable_grid, next_location, BOX),
+            lambda: next_variable_grid.at[tuple(next_location)]
+            .set(AGENT)
+            .at[tuple(box_location)]
+            .set(BOX),
+            lambda: next_variable_grid.at[tuple(next_location)].set(AGENT),
+        )
```

Comment:
NIT: slightly cleaner with `cond` or `where`.
```suggestion
        next_variable_grid = jax.lax.select(
            self.check_space(variable_grid, next_location, BOX),
            next_variable_grid.at[tuple(next_location)]
            .set(AGENT)
            .at[tuple(box_location)]
            .set(BOX),
            next_variable_grid.at[tuple(next_location)].set(AGENT),
        )
```